### PR TITLE
fix: corrected get last ar filed date method

### PR DESCRIPTION
**Pull request by EPortman**
*Issue #:* https://app.zenhub.com/workspaces/names-team-board-new-655554cbddd49510027dad2e/issues/gh/bcgov/business-ar/318

*Description of changes:*
When a Business AR filing is received from the new system, the corporation is looked up using `find_by_identifier`, which creates a founding_date based on the `recognition_date` (converted from Pacific Time to UTC).
https://github.com/bcgov/lear/blob/main/colin-api/src/colin_api/models/business.py#L327

When `_get_last_ar_filed_date` is called, it uses this founding_date by extracting the month and day. The issue was that the founding_date was not being converted back to Pacific Time. As a result, filings for businesses with a `recognition_date` after 5 PM local time were always showing a `last_ar_filed_date` one day ahead.

This PR fixes this by converting the `founding_date` back to pacific time before creating the `last_ar_filed_date`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: colin-api/src/colin_api/models/filing.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
     ShareObject,  # noqa: I001
 )  # noqa: I001
 from colin_api.resources.db import DB
-from colin_api.utils import convert_to_json_date, convert_to_json_datetime, convert_to_snake
+from colin_api.utils import convert_to_json_date, convert_to_json_datetime, convert_to_snake, convert_to_pacific_time
```
Comment (vysakh-menon-aot) [inline colin-api/src/colin_api/models/filing.py:48]:
```suggestion
from colin_api.utils import convert_to_json_date, convert_to_json_datetime, convert_to_pacific_time, convert_to_snake
```

Comment (EPortman) [reply colin-api/src/colin_api/models/filing.py]: Thanks!

**Comment by vysakh-menon-aot** [comment]
@EPortman If its ready you can change this from Draft to Ready for review and assign yourself as Assignees

**Comment by EPortman** [comment]
@vysakh-menon-aot I was getting an error when calling `convert_to_pacific_time()` since the `founding_date_str` had `-00:00` in it when it should be `+00.00`.
image



So just had to control for that.

**Comment by EPortman** [comment]
I also do not have the option to assign myself as the assignee. I think a permission issue?

**Comment by EPortman** [comment]
This is now ready. I tested it with a filing (Corp number `0705869`) and the result in Colin was like so:

image

The day and month are now correctly in sync with the `recogniton_date` when the time is after 5pm and the year matches the year of the AR filing.

File: colin-api/src/colin_api/models/filing.py [hunk 2]
```diff
@@ -1297,8 +1297,17 @@ def add_filing(cls, con, filing: Filing) -> int:
     @classmethod
     def _get_last_ar_filed_date(cls, header: dict, business: dict):
         filing_year = header.get('filingYear')
-        recognition_dt = datetime.datetime.fromisoformat(business.get('business').get('foundingDate')).date()
-        last_ar_filed_dt = f'{filing_year}-{recognition_dt.month}-{recognition_dt.day}'
+        founding_date_str = business.get('business').get('foundingDate')
+
+        # If the founding date has `-00:00`, replace it with `+00:00` before passing to the function
+        if '-00:00' in founding_date_str:
+            founding_date_str = founding_date_str.replace('-00:00', '+00:00')
```
Comment (vysakh-menon-aot) [inline colin-api/src/colin_api/models/filing.py:1304]: I am not really sure about this. Approving it since this function is used only for Business AR sync

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**